lib/rubygems/deprecate.rb

```diff
@@ -69,99 +69,101 @@
 #       end
 #     end
 
-module Gem::Deprecate
```

Comment:
Are those changes intended to be part of this PR? :thinking:

Reply:
Yes. In https://github.com/rubygems/rubygems/commit/73c199b087a94689a0e49b76e5333193f2310dcb, we made `rubygems/package` requirable independently. This is a solution to still make `rubygems/package` requirable independently without having to load the full `rubygems.rb` file from it, which is what's causing the circularity warning.

The "solution" is to make sure `Gem` module is defined explicitly instead of opening `Gem::Deprecate` directly. Apparently the Gem module placeholder being defined was the only thing needed for making `rubygems/package` independently.

We could also make this solution more explicit by explicitly defining the `Gem` module on top of `rubygems/package` before requiring anything?

Reply:
Ahh, that makes sense. All good, thanks for the explanation.